### ISSUE TITLE
Save environment during build

### ISSUE THREAD

**Issue by bryank-cs** [easybuilders/easybuild-framework#2798]
It would be very useful if a file was dropped either in /tmp/eb-* or buildpath that contains the environment variables as exported during the configure/build stages. This would allow sourcing of these to do some investigation of failed builds.

**Comment by boegel** [comment]
@bryank-cs Are you aware of the `--extended-dry-run` (or `-x`) option that spits out the largest part of the build environment that is set up by EasyBuild (i.e. the loaded modules + the environment variables set by the toolchain preparation step), in a way you can just copy-paste it.

See also https://easybuild.readthedocs.io/en/latest/Extended_dry_run.html.

It's not perfect, but it should also include environment variables set during `configure`/`build` steps & co...

**Comment by bryank-cs** [comment]
No, I was not aware of that. Not quite as good as just having it in a separate file, but workable. Thanks.
